### PR TITLE
feat: add TypeB plugin with API Key auth templates

### PR DESCRIPTION
**Pull request by huimiu**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/27822940

### CODE REVIEW

File: templates/js/api-plugin-from-scratch-bearer/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "compilerOptions": {
```
Comment (hund030) [inline templates/js/api-plugin-from-scratch-bearer/tsconfig.json:2]: tsconfig for js template?

Comment (huimiu) [reply templates/js/api-plugin-from-scratch-bearer/tsconfig.json:2]: Oh, I copied from the ts template, removed

**Review by eriolchan — APPROVED** [review]

File: templates/js/api-plugin-from-scratch-bearer/.vscode/launch.json
Comment (MuyangAmigo) [inline templates/js/api-plugin-from-scratch-bearer/.vscode/launch.json:1]: I think we are adding options to debug / preview in desktop, we probably should add to this template as well. FYI @xiaolang124

File: templates/js/api-plugin-from-scratch-bearer/README.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+# Overview of the Copilot Plugin template
+
+## Build a Copilot Plugin from a new API with Azure Functions
+
```
Comment (MuyangAmigo) [inline templates/js/api-plugin-from-scratch-bearer/README.md:4]: I suggest updating the title to include authentication mechanism here: `Build a Copilot Plugin from a new API authenticated by API Key with Azure Function`

**Pull request merged**